packages/snaps-controllers/src/interface/utils.ts

```diff
@@ -203,6 +275,50 @@ export function getDefaultAsset(
   };
 }
 
+/**
+ * Get the default state value for an account selector.
+ *
+ * @param element - The account selector element.
+ * @param elementDataGetters - Data getters for the element.
+ * @param elementDataGetters.getSelectedAccount - A function to get the selected account in the client.
+ * @param elementDataGetters.listAccounts - A function to list accounts for the provided chain IDs.
+ * @param elementDataGetters.snapOwnsAccount - A function to check if the snap owns the account.
+ * @returns The default state for the account selector.
+ */
+export function getAccountSelectorDefaultStateValue(
+  element: AccountSelectorElement,
+  { getSelectedAccount, listAccounts, snapOwnsAccount }: ElementDataGetters,
+) {
+  const { chainIds, hideExternalAccounts } = element.props;
+
+  const selectedAccount = getSelectedAccount();
+
+  if (
+    (!chainIds ||
+      chainIds.length === 0 ||
+      selectedAccount.scopes.some((scope) =>
+        isMatchingChainId(scope, chainIds),
+      )) &&
+    (!hideExternalAccounts ||
+      (hideExternalAccounts && snapOwnsAccount(selectedAccount)))
+  ) {
+    return formatAccountSelectorStateValue(selectedAccount, chainIds);
+  }
+
+  const accounts = listAccounts(chainIds);
+
+  const filteredAccounts = hideExternalAccounts
+    ? accounts.filter((account) => snapOwnsAccount(account))
+    : accounts;
+
+  assert(
+    filteredAccounts.length > 0,
+    'No accounts found for the provided chain IDs.',
+  );
```

Comment:
Is it safe to assert here? Wouldn't this crash the controller if a Snap then specifies an unsupported chain ID?

Reply:
I don't think I fully understand. Shouldn't the component have an empty state or something like the asset selector instead of throwing?

Reply:
There is no empty state as of right now, we could default to disabling the component if the value is `null`

Reply:
Hmm, what happens if a Snap passes an unknown address to `AssetSelector` for example? Do we also throw? I imagine we want to try and deal with these issues the same way

Reply:
We set the state to null and disable it in the UI: https://github.com/MetaMask/snaps/blob/b0b5264ffa924e2d3d3d24bb156d3009ca1d923a/packages/snaps-controllers/src/interface/utils.ts#L179-L183

I will do the same for the `AccountSelector`